src/main/scala/net/snowflake/spark/snowflake/io/StageReader.scala

```diff
@@ -36,13 +36,31 @@ private[snowflake] object StageReader {
 
     val prefix = Random.alphanumeric take 10 mkString ""
 
-    val res = buildUnloadStatement(
+    val copyStatement = buildUnloadStatement(
       params,
       statement,
       s"@$stage/$prefix/",
       compressFormat,
       format
-    ).execute(params.bindVariableEnabled)(conn)
+    )
+
+    val startTime = System.currentTimeMillis()
+    val (res, queryID, errorThrow) =
+      copyStatement.executeWithoutThrow(params.bindVariableEnabled)(conn)
```

Comment:
Why do we need this, aren't we logging telemetry in execute ?

Reply:
executeWithoutThrow() doesn't log error inside.
This is to sending a telemetry message when reading with COPY INTO LOCATION fails.
The telemetry message is independent to send log entries to telemetry.

PS: I am a little confusing about the question. If this doesn't address your comment, could you please explain the question?

Reply:
Oh, I see. I think you mean we can move to send the telemetry message inside of executeWithoutThrow().
It's doable, I didn't do that because some information for the telemetry message are collected by the caller.
If we want to send message inside of executeWithoutThrow(), we may add some extra parameters.
In the future, we will send successful messages too, some info are not ready in the executeWithoutThrow() function such as total data size, partition count, etc.
So, I think it should be better to process it by caller and keep executeWithoutThrow() simple.